packages/edit-site/src/components/page-pages/index.js

```diff
@@ -63,6 +63,9 @@ function useView( type ) {
 	const selectedDefaultView =
 		isCustom === 'false' &&
 		DEFAULT_VIEWS[ type ].find( ( { slug } ) => slug === activeView )?.view;
+	selectedDefaultView.layout = {
```

Comment:
This is to initialize the view with the layout info. It wasn't an issue until now because the default layout so far (table) doesn't have any data, and when the layout changed, it was updated.

There's an opportunity to simplify the existing logic further, to be explored in follow-ups.